modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadata.java

```diff
@@ -174,6 +188,16 @@ public Map<Integer, Set<Integer>> partitions() {
         return Collections.unmodifiableMap(locParts);
     }
 
+    /** */
+    public boolean isGroupWithCompresion(int grpId) {
+        return hasComprGrps && comprGrpIds.contains(grpId);
+    }
+
+    /** */
+    public boolean hasCompressedGroups() {
+        return hasComprGrps;
```

Comment:
Can we use F.isEmpty instead of a flag?

Reply:
Flag is better here, because of backward compatibility. We can add field and if field has not been serialized in older version, this flag will be false.